cmd/kubeadm/app/componentconfigs/kubelet_windows.go

```diff
@@ -57,7 +57,7 @@ func (kc *kubeletConfig) Mutate() error {
 
 func mutatePaths(cfg *kubeletconfig.KubeletConfiguration, drive string) {
 	mutateStringField := func(name string, field *string) {
-		// path.IsAbs() is not reliable here in the Windows runtime, so check if the
+		// filepath.IsAbs() is not reliable here in the Windows runtime, so check if the
 		// path starts with "/" instead. This means the path originated from a Unix node and
```

Comment:
The blow is using `!strings.HasPrefix(*field, "/")`, and the comment change seems to be wrong.

We should remove the comments here and change the below if statement to use `filepath.IsAbs()` according to https://github.com/golang/go/issues/1483.

Reply:
I disagree. The point in this comment is that for some reason ``/``-prefixed paths are not considered absolute paths in Windows golang, even though it is considered as an absolute path in other languages, and it is considered an absolute path (relative to the current drive letter) in Windows in general [1]. I have another PR open for something relate to absolute / relative paths. [2]

So, there are 2 outcomes of this function:

- the field starts with ``/``, and thus, an absolute path. In this case, it will prepend the drive letter to it.
- the field doesn't start with ``/``, in which case it is a noop.

If we'd use ``!filepath.IsAbs(*field)`` instead of ``!strings.HasPrefix(*field, "/"``, the 2 cases would instead do as follows:

- the field starts with ``/``, which technically is an absolute path, but ``filepath.IsAbs`` would return ``false``. This means that the function does nothing.
- the field doesn't start with ``/``, in which case ``filepath.IsAbs`` returns false, in which case it is a noop.

If we'd change ``strings.HasPrefix`` with ``filepath.IsAbs``, we'd have a regression.

[1] https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths
[2] https://github.com/kubernetes/kubernetes/pull/112566

Reply:
i can confirm windows had the abs problem in go's stdlib.